routes/entryRoutes.js

```diff
@@ -115,4 +118,31 @@ routes.GETstudentEntries = function(req, res){
   });
 }
 
+routes.GETcohortEntries = function(req, res) {
+  var cohort = req.params.cohort;
+  var studentListFull = [];
+  var attendanceList = [];
+  var starsList = [];
+  var datesList = [];
+  var behaviorList = [];
+  var warningList = [];
+  var engageContentList = []
+
+  Student.find({program:cohort}, function(err, students){
+    var studentIds = students.map(function(student){return student._id});
+    console.log('studentIds: '+studentIds);
+    FormDB.find({_studentID:{$in:studentIds}}, function(err, forms){
+      forms.reduce(function(prev,form){
+        attendanceList.push(form.attendance);
+        starsList.push(form.stars);
+        datesList.push(form.date);
+        warningList.push(form.warnings);
+        engageContentList.push(form.engageContent)
+      }, {})
```

Comment:
It is a syntax thing for reduce. What_ reduce _does is it loop through an array in an interesting way. It goes through each element of the array and at the end of each element, it calls the function (here it is the line function(prev, form)). It is sort of like a callback to get to the next element. Within the callback function, prev is the previous object and form is the next element to work on. Empty object is what the whole thing starts out with.